server/fleet/agent_options.go

```diff
@@ -269,6 +276,198 @@ type osqueryOptions struct {
 	YaraDelay                           uint32 `json:"yara_delay"`
 }
 
+// NOTE: generate automatically with `go run ./tools/osquery-agent-options/main.go`
```

Comment:
From @roperzh (was on #7986 but meant for this PR):

did you generate this on a Linux machine? I think there's a small edge case, which I'm unsure if we want to tackle or not: osquery has certain command line flags that are os-specific (search for "control flags" [here](https://osquery.readthedocs.io/en/stable/installation/cli-flags/))

Asking because I can see the Linux-specific hardware_disabled_types but I can't see the macOS ones

Reply:
I am on Linux, so indeed the os-specific flags are missing... Not sure how to address that with the auto-generated code though... :( We may have to add yet more manual steps...

Reply:
yes, I don't see a way other than adding those manually, and I just spotted there are windows-specific flags too.

whether is worth it or not, and how commonly used these are, I have no idea 🤔

Reply:
Really like your solution because it won’t fail independently of the OS in which the code is generated